prowler/lib/outputs/finding.py

```diff
@@ -0,0 +1,191 @@
+from datetime import datetime
+from enum import Enum
+from typing import Optional, Union
+
+from pydantic import BaseModel
+
+from prowler.config.config import prowler_version
+from prowler.lib.logger import logger
+from prowler.lib.outputs.common import (
+    fill_common_finding_data,
+    get_provider_data_mapping,
+)
+from prowler.lib.outputs.compliance.compliance import get_check_compliance
+
+
+class Status(str, Enum):
+    PASS = "PASS"
+    FAIL = "FAIL"
+    MANUAL = "MANUAL"
+
+
+class Severity(str, Enum):
+    critical = "critical"
+    high = "high"
+    medium = "medium"
+    low = "low"
+    informational = "informational"
+
+
+class Finding(BaseModel):
+    """
+    Finding generates a finding's output. It can be written to CSV or another format doing the mapping.
+
+    This is the base finding output model for every provider.
+    """
+
+    auth_method: str
+    timestamp: Union[int, datetime]
+    account_uid: str
+    # Optional since depends on permissions
```

Comment:
Subject required. Apply this to all other comments.

```suggestion
    # Optional since it depends on permissions
```

Reply:
Done